court/tests.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+from court.models import Court
+from decimal import Decimal
+from django.db.utils import IntegrityError
+from django.core.exceptions import ValidationError
+
+
+@pytest.mark.django_db
+class TestCourtModel:
+
+    @pytest.fixture
+    def court(self):
+        court = Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                                     city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+        return court
+
+    def test_validate_instance(self, court):
+        court_in_db = Court.objects.get(pk=court.pk)
+        assert court_in_db == court
+
+    def test_create_court_with_existing_id(self, court):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                                 city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30)
+
+    def test_create_valid_court(self):
+        Court.objects.create(courtID=1, x=Decimal('11'), y=Decimal('22'),
+                             city='TEST_CITY', neighborhood='TEST_NEIGHBORHOOD', max_players=30).full_clean()
+
+    def test_create_court_with_negative_max_players(self):
+        with pytest.raises(IntegrityError):
+            Court.objects.create(
+                x=Decimal('11'),
+                y=Decimal('22'),
+                city='TEST_CITY',
+                neighborhood='TEST_NEIGHBORHOOD',
+                max_players=-100
+            ).full_clean()
```

Comment:
Is this exception raised by the call to full_clean (I think  the summary of the conversation from yesterday was that it is not)? If not, then the test should pass if you remove the call to full_clean.

Reply:
You're right, I've totally forgot to delete the `.full_clean()` cmd. The `IntegrityError` occurs without it.